demo/corpus/logic/src/embedding/util.ts

```diff
@@ -0,0 +1,29 @@
+/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
+PDX-License-Identifier: Apache-2.0 */
+
+import { IEmbeddingModelInfo } from '@aws/galileo-sdk/lib/models';
+import { AsyncCallerParams } from 'langchain/dist/util/async_caller';
+import { Embeddings } from 'langchain/embeddings/base';
+import { SageMakerEndpointEmbeddings } from '.';
+import { ENV } from '../env';
+
+// A map to store the mappings between embedding model id and embeddings
+const __EMBEDDINGS_CACHE__ = new Map<string, Embeddings>();
+
+export function getEmbeddingsByModelId(embeddingModelId: string, params?: AsyncCallerParams): Embeddings {
+  if (__EMBEDDINGS_CACHE__.has(embeddingModelId)) {
+    return __EMBEDDINGS_CACHE__.get(embeddingModelId)!;
+  } else {
+    const embeddings = new SageMakerEndpointEmbeddings(params ?? {}, embeddingModelId);
+    __EMBEDDINGS_CACHE__.set(embeddingModelId, embeddings);
+    return embeddings;
+  }
+}
+
+export function findEmbeddingModelByRefKey(modelRefKey?: string): IEmbeddingModelInfo {
+  const embeddingModels = ENV.EMBEDDINGS_SAGEMAKER_MODELS;
+
+  if (!modelRefKey) return embeddingModels[0];
```

Comment:
Regarding `make sure if modelRefKey not supplied, the default model is returned and not the first one`, the embedding models are sorted in CorpusStack.ts, and the default model is always placed as the fist element in the array. Will address the first and third bullet points in the next cr.

Reply:
Regarding `make sure this array has at least one element. in env.ts the fallback is []`, this is ensured in
https://github.com/aws-samples/aws-genai-conversational-rag-reference/blob/mainline/demo/infra/src/application/corpus/index.ts#L76. In the env.ts, the fall back logic `[]` is never going to hit. It is just used to prevent linting issue.